src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp

```diff
@@ -38,59 +38,60 @@
 
 using namespace Antares::Data;
 
+
+void exportPaliers(PROBLEME_HEBDO* problemeHebdo,
+                    CORRESPONDANCES_DES_VARIABLES* CorrespondanceVarNativesVarOptim,
+                    int pays, int* nombreDeTermes,
+                    double* Pi, int* Colonne,
+                    int timeStepInYear)
+{
+    const PALIERS_THERMIQUES* PaliersThermiquesDuPays = problemeHebdo->PaliersThermiquesDuPays[pays];
+    int nvars = problemeHebdo->ProblemeAResoudre->NombreDeVariables;
+    std::vector<std::string> varname;
+    varname.assign(nvars, "");
```

Comment:
`varname` should be passed by reference. Otherwise `OPT_Export_add_variable` will just write to a temporary variable, which is useless.